ui/app/components/app/transaction-list/transaction-list.component.js

```diff
@@ -5,20 +5,31 @@ import {
   nonceSortedCompletedTransactionsSelector,
   nonceSortedPendingTransactionsSelector,
 } from '../../../selectors/transactions';
+import { getCurrentChainId } from '../../../selectors';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import TransactionListItem from '../transaction-list-item';
 import Button from '../../ui/button';
 import { TOKEN_CATEGORY_HASH } from '../../../helpers/constants/transactions';
-import { SWAPS_CONTRACT_ADDRESS } from '../../../../../shared/constants/swaps';
+import { SWAPS_CHAINID_CONTRACT_ADDRESS_MAP } from '../../../../../shared/constants/swaps';
 import { TRANSACTION_TYPES } from '../../../../../shared/constants/transaction';
 
 const PAGE_INCREMENT = 10;
 
-const getTransactionGroupRecipientAddressFilter = (recipientAddress) => {
+// When we are on a token page, we only want to show transactions that involve that token.
+// In the case of token transfers or approvals, these will be transactions sent to the
+// token contract. In the case of swaps, these will be transactions sent to the swaps contract
+// and which have the token address in the transaction data.
+//
+// getTransactionGroupRecipientAddressFilter is used to determine whether a transaction matches
+// either of those criteria
+const getTransactionGroupRecipientAddressFilter = (
+  recipientAddress,
+  chainId,
+) => {
   return ({ initialTransaction: { txParams } }) => {
     return (
       txParams?.to === recipientAddress ||
```

Comment:
Should we do a check here for:

```
if (!txParams) {  return ____ }
```

Might be shorter than a bunch of `?.` optional chaining checks.

Reply:
Given our lint rules, this would have to be over 3 lines, meanwhile two other lines just become shorter by a single character. So might be best to leave as is, unless adding the explicit `if` check and return is a readability improvement?